Sources/NIO/SocketOptionProvider.swift

```diff
@@ -70,9 +104,20 @@ public protocol SocketOptionProvider {
     ///     - name: The name of the socket option, e.g. `SO_REUSEADDR`.
     /// - returns: An `EventLoopFuture` containing the value of the socket option, or
     ///     any error that occurred while retrieving the socket option.
-    func unsafeGetSocketOption<Value>(level: SocketOptionLevel, name: SocketOptionName) -> EventLoopFuture<Value>
+    func unsafeGetSocketOption<Value>(level: NIOBSDSocket.OptionLevel, name: NIOBSDSocket.Option) -> EventLoopFuture<Value>
 }
 
+#if !os(Windows)
+    extension SocketOptionProvider {
+        func unsafeSetSocketOption<Value>(level: SocketOptionLevel, name: SocketOptionName, value: Value) -> EventLoopFuture<Void> {
+            return unsafeSetSocketOption(level: NIOBSDSocket.OptionLevel(rawValue: CInt(level)), name: NIOBSDSocket.Option(rawValue: CInt(name)), value: value)
+        }
+
+        func unsafeGetSocketOption<Value>(level: SocketOptionLevel, name: SocketOptionName) -> EventLoopFuture<Value> {
+            return unsafeGetSocketOption(level: NIOBSDSocket.OptionLevel(rawValue: CInt(level)), name: NIOBSDSocket.Option(rawValue: CInt(name)))
+        }
+    }
```

Comment:
Unfortunately these are backwards: we need default implementations for the _new_ protocol requirements, in terms of the _old_ ones.

The goal is that, if someone else has conformed their type to `SocketOptionProvider`, that their type does not break when they update NIO. To achieve that, there must be no new protocol requirements that do not have default implementations.

Reply:
Ah, good point.  Inverted.